### DISCUSSION TITLE
How to install "openjdk-8-jdk" or "JDK 1.8" in Dockfile?

### DISCUSSION THREAD

**Discussion by JenniferYingyiWu2020** [tiangolo/uvicorn-gunicorn-fastapi-docker#306] [Questions] [answered]
Dear All,
    I have read the "How to use" and "Quick Start" section of your README.md file, and have created the Dockerfile using 
"
FROM tiangolo/uvicorn-gunicorn-fastapi:python3.7
" 
![11](https://user-images.githubusercontent.com/72843232/96842942-4d4a0c80-1480-11eb-9408-ab1d59a56aff.png)

    Due to "openjdk-8-jdk" is needed in my project, and I searched "JDK installation package of the docker image" when the command "docker build -t ..." have been executed:
"
RUN apt-get update && apt search jdk
"
    However, instead of "openjdk-8-jdk", just only "openjdk-11-jdk" was found. Because of "openjdk-8-jdk" is the only right version of "JDK" that can be used in my project, could you pls help to give me some suggestions for how to install "openjdk-8-jdk" or "jdk 1.8" in the Dockerfile? Thanks!

    Some log information of "docker build -t ..." is below: (The log file is only related with "apt search jdk")
[search_jdk.txt](https://github.com/tiangolo/uvicorn-gunicorn-fastapi-docker/files/5421105/search_jdk.txt)

**Comment by graue70** [comment]
See [this guide](https://guides.github.com/features/mastering-markdown/) for information on how to format your post. There is also a preview feature where you can check your text before posting it.

**Comment by graue70** [comment, marked as answer]
Regarding your issue: `uvicorn-gunicorn-fastapi:python3.7` is based on `uvicorn-gunicorn:python3.7` which is based on `python:3.7` which is based on `buildpack-deps:buster` which is a Debian Buster OS.
That means you can install `openjdk-8-jdk` in `uvicorn-gunicorn-fastapi:python3.7` the same way you can install it on Debian Buster. You should be able to find information on how to do that.
